### PR TITLE
JDK-8300241: Replace NULL with nullptr in share/classfile/

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory share/classfile/. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in
a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300241](https://bugs.openjdk.org/browse/JDK-8300241): Replace NULL with nullptr in share/classfile/

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [71a7162b](https://git.openjdk.org/jdk/pull/12030/files/71a7162bafde56bf738bbaee22982d6c2abb2e19)
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12030/head:pull/12030` \
`$ git checkout pull/12030`

Update a local copy of the PR: \
`$ git checkout pull/12030` \
`$ git pull https://git.openjdk.org/jdk pull/12030/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12030`

View PR using the GUI difftool: \
`$ git pr show -t 12030`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12030.diff">https://git.openjdk.org/jdk/pull/12030.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/classfile/classFileParser.cpp [hunk 1]
```diff
@@ -793,12 +793,12 @@ static void initialize_hashtable(NameSigHash** table) {
 // NOTE: caller should guarantee that GC doesn't happen during the life cycle
 // of table since we don't expect Symbol*'s to move.
 static bool put_after_lookup(const Symbol* name, const Symbol* sig, NameSigHash** table) {
-  assert(name != NULL, "name in constant pool is NULL");
+  assert(name != nullptr, "name in constant pool is nullptr");
```
Comment (jdksjolen) [inline src/hotspot/share/classfile/classFileParser.cpp:796]: Change to null in string!

File: src/hotspot/share/classfile/classFileParser.cpp [hunk 2]
```diff
@@ -886,14 +886,14 @@ void ClassFileParser::parse_interfaces(const ClassFileStream* const stream,
...
-    const Symbol* name = NULL;
+    const Symbol* name = nullptr;
     {
       debug_only(NoSafepointVerifier nsv;)
       for (index = 0; index < itfs_len; index++) {
         const InstanceKlass* const k = _local_interfaces->at(index);
         name = k->name();
-        // If no duplicates, add (name, NULL) in hashtable interface_names.
-        if (!put_after_lookup(name, NULL, interface_names)) {
+        // If no duplicates, add (name, null) in hashtable interface_names.
```
Comment (jdksjolen) [inline src/hotspot/share/classfile/classFileParser.cpp:895]: Maybe nullptr better here?

File: src/hotspot/share/classfile/javaClasses.cpp [hunk 3]
```diff
@@ -4472,19 +4472,19 @@ int  java_lang_ClassLoader::_unnamedModule_offset;
 int  java_lang_ClassLoader::_parent_offset;
 
 ClassLoaderData* java_lang_ClassLoader::loader_data_acquire(oop loader) {
-  assert(loader != NULL, "loader must not be NULL");
+  assert(loader != nullptr, "loader must not be nullptr");
```
Comment (jdksjolen) [inline src/hotspot/share/classfile/javaClasses.cpp:4475]: Fix nullptr in string

File: src/hotspot/share/classfile/klassFactory.hpp [hunk 4]
```diff
@@ -43,17 +43,17 @@ class Symbol;
...
  *   (no pending exceptions)
  *
  * Returns:
- *   if the returned value is non-NULL, that value is an indirection (pointer/handle)
+ *   if the returned value is non-nullptr, that value is an indirection (pointer/handle)
  *   to a Klass. The caller will not have a pending exception.
  *
  *   On broken invariants and/or runtime errors the returned value will be
- *   NULL (or a NULL handle) and the caller *might* now have a pending exception.
+ *   nullptr (or a nullptr handle) and the caller *might* now have a pending exception.
```
Comment (jdksjolen) [inline src/hotspot/share/classfile/klassFactory.hpp:56]: Fix nullptr in comments.

File: src/hotspot/share/classfile/modules.cpp [hunk 5]
```diff
@@ -427,18 +427,18 @@ void Modules::define_module(Handle module, jboolean is_open, jstring version,
...
-                         location_symbol != NULL ? location_symbol->as_C_string() : "NULL");
+                         location_symbol != nullptr ? location_symbol->as_C_string() : "nullptr");
   LogTarget(Debug, module) lt;
   if (lt.is_enabled()) {
     LogStream ls(lt);
     ls.print("define_module(): creation of module: %s, version: %s, location: %s, ",
-                 module_name, version_symbol != NULL ? version_symbol->as_C_string() : "NULL",
-                 location_symbol != NULL ? location_symbol->as_C_string() : "NULL");
+                 module_name, version_symbol != nullptr ? version_symbol->as_C_string() : "nullptr",
+                 location_symbol != nullptr ? location_symbol->as_C_string() : "nullptr");
```
Comment (jdksjolen) [inline src/hotspot/share/classfile/modules.cpp:441]: Fix nullptr in string

File: src/hotspot/share/classfile/resolutionErrors.cpp [hunk 6]
```diff
@@ -76,7 +76,7 @@ void ResolutionErrorTable::add_entry(const constantPoolHandle& pool, int cp_inde
                                      const char* message)
 {
   assert_locked_or_safepoint(SystemDictionary_lock);
-  assert(!pool.is_null() && message != NULL, "adding NULL obj");
+  assert(!pool.is_null() && message != nullptr, "adding nullptr obj");
```
Comment (jdksjolen) [inline src/hotspot/share/classfile/resolutionErrors.cpp:79]: Fix nullptr in string

File: src/hotspot/share/classfile/resolutionErrors.cpp [hunk 7]
```diff
@@ -64,7 +64,7 @@ void ResolutionErrorTable::add_entry(const constantPoolHandle& pool, int cp_inde
                                      Symbol* cause, Symbol* cause_msg)
 {
   assert_locked_or_safepoint(SystemDictionary_lock);
-  assert(!pool.is_null() && error != NULL, "adding NULL obj");
+  assert(!pool.is_null() && error != nullptr, "adding nullptr obj");
```
Comment (jdksjolen) [inline src/hotspot/share/classfile/resolutionErrors.cpp:67]: Fix nullptr in string

File: src/hotspot/share/classfile/systemDictionary.cpp [hunk 8]
```diff
@@ -518,7 +518,7 @@ static void double_lock_wait(JavaThread* thread, Handle lockObject) {
   assert(EnableWaitForParallelLoad,
          "Only called when enabling legacy parallel class loading logic "
          "for non-parallel capable class loaders");
-  assert(lockObject() != NULL, "lockObject must be non-NULL");
+  assert(lockObject() != nullptr, "lockObject must be non-nullptr");
```
Comment (jdksjolen) [inline src/hotspot/share/classfile/systemDictionary.cpp:521]: Fix nullptr in string

File: src/hotspot/share/classfile/verificationType.cpp [hunk 9]
```diff
@@ -187,10 +187,10 @@ void VerificationType::print_on(outputStream* st) const {
       } else if (is_uninitialized()) {
         st->print("uninitialized %d", bci());
       } else {
-        if (name() != NULL) {
+        if (name() != nullptr) {
           name()->print_value_on(st);
         } else {
-          st->print_cr("NULL");
+          st->print_cr("nullptr");
         }
```
Comment (jdksjolen) [inline src/hotspot/share/classfile/verificationType.cpp:194]: Fix nullptr in string

File: src/hotspot/share/classfile/verifier.cpp [hunk 10]
```diff
@@ -2833,10 +2833,10 @@ void ClassVerifier::verify_invoke_instructions(
   // Get the signature's verification types.
   sig_as_verification_types* mth_sig_verif_types;
   sig_as_verification_types** mth_sig_verif_types_ptr = method_signatures_table()->get(sig_index);
-  if (mth_sig_verif_types_ptr != NULL) {
+  if (mth_sig_verif_types_ptr != nullptr) {
     // Found the entry for the signature's verification types in the hash table.
     mth_sig_verif_types = *mth_sig_verif_types_ptr;
-    assert(mth_sig_verif_types != NULL, "Unexpected NULL sig_as_verification_types value");
+    assert(mth_sig_verif_types != nullptr, "Unexpected nullptr sig_as_verification_types value");
```
Comment (jdksjolen) [inline src/hotspot/share/classfile/verifier.cpp:2839]: Fix nullptr in string

**Review by jdksjolen — COMMENTED** [review]
Found some manual fix ups.

**Comment by jdksjolen** [comment]
Hi, I'm opening this up for review. It builds on my machine and I'm sending it off for tier1 testing. I haven't fixed my comments yet.

File: src/hotspot/share/classfile/verifier.cpp (hunk 10 above)
Comment (coleenp) [inline src/hotspot/share/classfile/verifier.cpp:2839]:
```suggestion
    assert(mth_sig_verif_types != nullptr, "Unexpected null sig_as_verification_types value");
```

**Review by coleenp — APPROVED** [review]
I didn't see any other places that needed changing.

File: src/hotspot/share/classfile/classFileParser.cpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/share/classfile/classFileParser.cpp:796]:
```suggestion
  assert(name != nullptr, "name in constant pool is null");
```

File: src/hotspot/share/classfile/classFileParser.cpp (hunk 2 above)
Comment (coleenp) [reply src/hotspot/share/classfile/classFileParser.cpp:895]:
```suggestion
        // If no duplicates, add (name, nullptr) in hashtable interface_names.
```

File: src/hotspot/share/classfile/javaClasses.cpp (hunk 3 above)
Comment (coleenp) [reply src/hotspot/share/classfile/javaClasses.cpp:4475]:
```suggestion
  assert(loader != nullptr, "loader must not be null");
```

File: src/hotspot/share/classfile/javaClasses.cpp [hunk 11]
```diff
@@ -4472,19 +4472,19 @@ int  java_lang_ClassLoader::_unnamedModule_offset;
...
   assert(oopDesc::is_oop(loader), "loader must be oop");
   return Atomic::load_acquire(loader->field_addr<ClassLoaderData*>(_loader_data_offset));
 }
 
 ClassLoaderData* java_lang_ClassLoader::loader_data(oop loader) {
-  assert(loader != NULL, "loader must not be NULL");
+  assert(loader != nullptr, "loader must not be nullptr");
```
[the remark this answers is not in the document]
Comment (coleenp) [reply src/hotspot/share/classfile/javaClasses.cpp:4481]:
```suggestion
  assert(loader != nullptr, "loader must not be null");
```

File: src/hotspot/share/classfile/javaClasses.cpp [hunk 12]
```diff
@@ -4472,19 +4472,19 @@ int  java_lang_ClassLoader::_unnamedModule_offset;
...
   assert(oopDesc::is_oop(loader), "loader must be oop");
   return *loader->field_addr<ClassLoaderData*>(_loader_data_offset);
 }
 
 void java_lang_ClassLoader::release_set_loader_data(oop loader, ClassLoaderData* new_data) {
-  assert(loader != NULL, "loader must not be NULL");
+  assert(loader != nullptr, "loader must not be nullptr");
```
[the remark this answers is not in the document]
Comment (coleenp) [reply src/hotspot/share/classfile/javaClasses.cpp:4487]:
```suggestion
  assert(loader != nullptr, "loader must not be null");
```

File: src/hotspot/share/classfile/modules.cpp [hunk 13]
```diff
@@ -54,20 +54,20 @@
...
 
 static bool verify_module_name(const char *module_name, int len) {
-  assert(module_name != NULL, "invariant");
+  assert(module_name != nullptr, "invariant");
   return (len > 0 && len <= Symbol::max_length());
 }
 
 static bool verify_package_name(const char* package_name, int len) {
-  assert(package_name != NULL, "Package name derived from non-null jstring can't be NULL");
+  assert(package_name != nullptr, "Package name derived from non-null jstring can't be nullptr");
```
[the remark this answers is not in the document]
Comment (coleenp) [reply src/hotspot/share/classfile/modules.cpp:62]:
```suggestion
  assert(package_name != nullptr, "Package name derived from non-null jstring can't be null");
```

**Comment by dholmes-ora** [comment]
I will wait for the update before reviewing :)

**Comment by jdksjolen** [comment]
Hi @coleenp , I didn't click your suggested changes as they seem to be creating single commits for each change!

**Comment by coleenp** [comment]
> Hi @coleenp , I didn't click your suggested changes as they seem to be creating single commits for each change!
I think they have to be all or nothing to work, but it might be a good way to review these.

**Comment by jdksjolen** [comment]
Passes tier1.

**Review by iklam — APPROVED** [review]

**Comment by jdksjolen** [comment]
/integrate

Still passes tier1

**Pull request closed without merging**